apps/bmd/src/pages/PollWorkerScreen.tsx

```diff
@@ -182,6 +219,83 @@ const PollWorkerScreen: React.FC<Props> = ({
     )
   }
 
+  const isMachineTallySaved =
+    talliesOnCard &&
+    !!talliesOnCard.metadata.find(
+      (metadata) => metadata.machineId === machineConfig.machineId
+    )
+
+  const combinedTally =
+    talliesOnCard === undefined
+      ? tally
+      : isMachineTallySaved
+      ? talliesOnCard.tally
+      : combineTallies(election, talliesOnCard.tally, tally)
+
+  const combinedBallotsPrinted =
+    talliesOnCard === undefined
+      ? ballotsPrintedCount
+      : isMachineTallySaved
+      ? talliesOnCard.totalBallotsPrinted
+      : talliesOnCard.totalBallotsPrinted + ballotsPrintedCount
+
+  const handleSavingTally = async () => {
+    setIsSavingTally(true)
+    const metadata = talliesOnCard?.metadata.slice() ?? []
+    metadata.push({
+      machineId: machineConfig.machineId,
+      timeSaved: Date.now(),
+      ballotsPrinted: ballotsPrintedCount,
+    })
+    await saveTallyToCard({
+      tally: combinedTally,
+      metadata,
+      totalBallotsPrinted: combinedBallotsPrinted,
+    })
+    setIsSavingTally(false)
+  }
+  const metadataRows: React.ReactChild[] = []
+  const machineMetadata = talliesOnCard?.metadata.slice() ?? []
+  if (!isMachineTallySaved) {
+    metadataRows.push(
+      <tr key={machineConfig.machineId} data-testid="tally-machine-row">
+        <td>
+          {machineConfig.machineId}
+          {' (current machine)'}
+        </td>
+        <td>
+          <Button small onPress={handleSavingTally}>
+            Save to Card
+          </Button>
+        </td>
+      </tr>
+    )
+    machineMetadata.push({
+      machineId: machineConfig.machineId,
+      timeSaved: Date.now(),
+      ballotsPrinted: ballotsPrintedCount,
+    })
+  }
+  if (talliesOnCard !== undefined) {
+    metadataRows.push(
+      ...talliesOnCard.metadata
+        .slice()
+        .sort((a, b) => b.timeSaved - a.timeSaved)
```

Comment:
I believe this is a safe `sort` because of the `...` expansion, right? /cc @eventualbuddha

Reply:
It's safe because of the `.slice()`, which makes a copy. Another way would have been `[...talliesOnCard.metadata].sort()`, which is probably what you're thinking of @benadida.